### PR TITLE
InterpolateAtPoint near borders

### LINKED ISSUE #10501: InterpolateAtPoint is not continuous at the borders
### What is the bug?

`GDALRasterInterpolateAtPoint` is using the interpolation algorithm from RPC (that suffers the same behavior).
In the borders the interpolations are downgraded from CubicSpline to Bilinear, and from Bilinear to Nearest.
You can see an example here, using the image from https://en.wikipedia.org/wiki/Bicubic_interpolation (see that we use CubicSpline, that is a bit different)
old
In the image there are cross-sections where you can see the discontinuities.

I am working in a PR to "extend" the image in the borders, assuming a constant pixel value, and let the interpolation compute the values.
It should look something like this:
new

As a bonus it will contain **Cubic** interpolation (in addition to CubicSpline) ;)

### Steps to reproduce the issue

Request values in the borders of the image (like at [0 - 0.5) for bilinear or [0 - 1.5) for cubicspline)

### Versions and provenance

Tested in Ubuntu 20.04, but happens on every platform. For the RPC interpolation, it is happening the same.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jjimenezshaw**
## What does this PR do?
Makes `GDALRasterInterpolateAtPoint` continuous near the borders of the image.
It also adds `cubic` interpolation.
cubic
(why adding cubic here? Unlike cubicspline, cubic matches the values of the grid in the nodes. That makes testing during development easier. See that in the images at x=0.5 and y=0.5, that are cutting at the nodes of the grid).
See that cubic may get values bigger than the maximum of the grid values.

### Implementation
How is it getting data near the borders, if there are no point to interpolate with? It "extends" the image with the pixel value at the border.

For instance, this is how the top left corner is "extended" to have a 4x4 kernel:
```
.___      a a a b
|a b  =>  a a a b
|c i      a a a b
          c c c i
```

## What are related issues/pull requests?
Fixes #10501

## Tasklist

 - [x] Make sure code is correctly formatted (cf [pre-commit configuration](https://gdal.org/development/dev_practices.html#commit-hooks))
 - [x] Add test case(s)
 - [x] Add documentation
 - [x] Updated Python API documentation (swig/include/python/docs/)
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

**Comment by jjimenezshaw** [comment]
fyi @mdsumner @rouault

File: alg/gdal_interpolateatpoint.cpp [hunk 1]
```diff
@@ -136,6 +136,28 @@ GDALInterpExtractDEMWindow(GDALRasterBand *pBand,
     return true;
 }
 
+static double CubicKernel(double dfX)
```
Comment (jjimenezshaw) [inline alg/gdal_interpolateatpoint.cpp:139]: Should this function be in the header file? Currently is not used anywhere else. It is copied from `alg/gdalwarpkernel.cpp`

Comment (rouault) [reply alg/gdal_interpolateatpoint.cpp:139]: > Should this function be in the header file

we should try to limit copy&paste. Perhaps as a inline function in a alg/gdalresamplingkernels.h header used by gdal_interpolateatpoint.cpp and gdalwarpkernel.cpp (I also considered in the existing alg/gdal_alg_priv.h). Or any equivalent suggestion you may have

Comment (jjimenezshaw) [reply alg/gdal_interpolateatpoint.cpp]: done

**Comment by jjimenezshaw** [comment]
There is a persistent failing test `test_vsiaz_write_blockblob_chunk_size_1` https://github.com/OSGeo/gdal/actions/runs/10130627537/job/28012234353
that seems to be networking related. Not with my changes I guess.
Is it fixed on master? should I rebase in any case?

**Comment by rouault** [comment]
> Is it fixed on master?

I've just pushed a commit to skip it on CI. It is indeed quite flaky especially on this macos target.

>  should I rebase in any case?

no, I'll merge it as it

**Pull request merged**